File: Site/Models/BurndownDataPoint.cs
using System;
using System.Text.RegularExpressions;
using System.Globalization;

namespace CardWall.Models
{
    public struct BurndownDataPoint 
    {
        const char Separator = ';';
        public readonly DateTime Date;
        public readonly int PointsRemaining;

        public BurndownDataPoint(DateTime date, int pointsRemaining) {
            this.Date = date;
            this.PointsRemaining = pointsRemaining;
        }

        public override string ToString() {
            return string.Format("{0}{1}{2}", Date.ToShortDateString(), Separator, PointsRemaining);
        }

        public static BurndownDataPoint Parse(string input) {
            var parts = input.Split(new[]{ Separator }, StringSplitOptions.RemoveEmptyEntries);
            return new BurndownDataPoint(DateTime.Parse(parts[0], CultureInfo.InvariantCulture), int.Parse(parts[1]));
        }
    }
}

File: Source/BurndownDataPoint.fs
namespace CardWall
open System
open System.Globalization

type BurndownDataPoint(date, pointsRemaining, totalPointsBurned) =
    [<Literal>]
    static let Separator = ';'

    member this.Date : DateTime = date
    member this.PointsRemaining : int = pointsRemaining
    member this.TotalPointsBurned : int = totalPointsBurned

    static member Parse(input:string) =
        let parts = input.Split([| Separator |])
        BurndownDataPoint(DateTime.Parse(parts.[0], CultureInfo.InvariantCulture), int(parts.[1]), int(parts.[2]))

    override this.ToString() =
        String.Format("{0}{1}{2}{1}{3}", date.ToShortDateString(), Separator, pointsRemaining, totalPointsBurned);